File: StringGen/csharp/StringGen/Domain/StringGenerator.cs
using System;

namespace StringGen.Domain
{
    //Uses a Random generator to produce a string composed of characters from a given set.
    public class StringGenerator : IStringGenerator
    {
        public string GenerateString(string characterSet, int length)
        {
            var random = new Random();
            var result = new char[length];
            for (int i = 0; i < length; i++)
            {
                int index = random.Next(characterSet.Length);
                result[i] = characterSet[index];
            }
            return new string(result);
        }
    }
}

File: sim/java/stringgen/src/main/java/com/stringgen/domain/StringGenerator.java
package com.stringgen.domain;

import java.util.Random;

//Implements IStringGenerator to produce a random string from a given character set.
public class StringGenerator implements IStringGenerator {

    //Generates a random string using a Random generator and the provided characterSet.
    @Override
    public String generateString(String characterSet, int length) {
        Random random = new Random();
        StringBuilder sb = new StringBuilder(length);

        for (int i = 0; i < length; i++) {
            int index = random.nextInt(characterSet.length());
            sb.append(characterSet.charAt(index));
        }

        return sb.toString();
    }
}
